### PR TITLE
feat: add sortingStrategy param to connect, deprecate skipPermitationparam in composeTransaction

### PR DESCRIPTION
**Pull request by peter-sanderson**
Part of https://github.com/trezor/trezor-suite/issues/10765

### CODE REVIEW

File: packages/utxo-lib/tests/__fixtures__/compose.ts [hunk 1]
```diff
@@ -1059,80 +1059,6 @@ export const composeTxFixture = [
             type: 'final',
         },
     },
-    {
-        description:
-            'skip inputs/outputs permutation when skipPermutation=true (compatibility API)',
```
Comment (peter-sanderson) [inline packages/utxo-lib/tests/__fixtures__/compose.ts:1064]: there is already test `description: 'skip inputs/outputs permutation when sortingStrategy=none',`

File: packages/connect/src/api/bitcoin/TransactionComposer.ts [hunk 2]
```diff
@@ -10,14 +10,15 @@ import type {
     ComposedInputs,
     ComposeResult,
 } from '../../types/api/composeTransaction';
+import { TransactionInputOutputSortingStrategy } from '@trezor/utxo-lib/libDev/src/types';
```
Comment (martykan) [inline packages/connect/src/api/bitcoin/TransactionComposer.ts]:
```suggestion
import { TransactionInputOutputSortingStrategy } from '@trezor/utxo-lib/src/types';
```

File: packages/connect/e2e/__fixtures__/composeTransaction.ts [hunk 3]
```diff
@@ -113,7 +113,7 @@ export default {
                     },
                 ],
                 baseFee: 167,
-                skipPermutation: true,
+                sortingStrategy: 'none',
```
Comment (mroz22) [inline packages/connect/e2e/__fixtures__/composeTransaction.ts:150]: maybe duplicate this fixture and keep it to test legacy API?

File: packages/connect/src/api/composeTransaction.ts [hunk 4]
```diff
@@ -157,7 +170,8 @@ export default class ComposeTransaction extends AbstractMethod<'composeTransacti
             coinInfo,
             outputs,
             baseFee,
-            skipPermutation,
+            sortingStrategy:
+                skipPermutation === true ? 'none' : sortingStrategy ?? DEFAULT_SORTING_STRATEGY,
```
Comment (mroz22) [inline packages/connect/src/api/composeTransaction.ts]: do we need to care about `skipPermutation` here? I am not sure but I would expect this legacy param is already translated to the current param in `init`?

Comment (peter-sanderson) [reply packages/connect/src/api/composeTransaction.ts]: fixed

File: packages/connect/e2e/__fixtures__/composeTransaction.ts (hunk 3 above)
Comment (peter-sanderson) [reply packages/connect/e2e/__fixtures__/composeTransaction.ts:150]: fixed

**Review by mroz22 — APPROVED** [review]
lgtm 🏳️‍🌈

**Pull request merged**